source/Halibut.TestProxy/HttpProxyService.cs

```diff
@@ -219,10 +219,10 @@ async Task HandleProxyRequest(TcpClient client, CancellationToken cancellationTo
             }
         }
 
-        async Task CreateProxyConnectionAndForward(TcpClient client, HttpProxyConnectionRequest connectionRequest, CancellationToken cancellationToken)
+        void CreateProxyConnectionAndForward(TcpClient client, HttpProxyConnectionRequest connectionRequest, CancellationToken cancellationToken)
         {
             var proxyConnection = proxyConnectionService.GetProxyConnection(connectionRequest.Endpoint);
-            await proxyConnection.Connect(client, cancellationToken);
+            proxyConnection.Connect(client, cancellationToken);
```

Comment:
We don't want this to be async?

Reply:
Maybe. But the reason I removed it completely was because:
- It didn't actually do anything that was async
- Therefore it's misleading, as you expect it to run on another thread, but in reality, .NET is clever enough to run the code in its original thread until is reaches something it has to await

So I thought it more clear to call it what it was (not async)